lib/ui/painting/fragment_shader.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <iostream>
+
+#include "flutter/lib/ui/painting/fragment_shader.h"
+
+#include "flutter/lib/ui/dart_wrapper.h"
+#include "flutter/lib/ui/ui_dart_state.h"
+#include "third_party/skia/include/core/SkString.h"
+#include "third_party/tonic/converter/dart_converter.h"
+#include "third_party/tonic/dart_args.h"
+#include "third_party/tonic/dart_binding_macros.h"
+#include "third_party/tonic/dart_library_natives.h"
+#include "third_party/tonic/typed_data/typed_list.h"
+
+using tonic::ToDart;
+
+namespace flutter {
+
+static void FragmentShader_constructor(Dart_NativeArguments args) {
+  DartCallConstructor(&FragmentShader::Create, args);
+}
+
+IMPLEMENT_WRAPPERTYPEINFO(ui, FragmentShader);
+
+#define FOR_EACH_BINDING(V) \
+  V(FragmentShader, init)   \
+  V(FragmentShader, update)
+
+FOR_EACH_BINDING(DART_NATIVE_CALLBACK)
+
+void FragmentShader::RegisterNatives(tonic::DartLibraryNatives* natives) {
+  natives->Register(
+      {{"FragmentShader_constructor", FragmentShader_constructor, 1, true},
+       FOR_EACH_BINDING(DART_REGISTER_NATIVE)});
+}
+
+sk_sp<SkShader> FragmentShader::shader(SkSamplingOptions sampling) {
+  // TODO(insert github issue): implement this method.
```

Comment:
It looks like this needs attention.

Reply:
I added a tracker for this. Right now we are ignoring sampling options. If you think it should be fully implemented in this PR, should it basically follow what image_shader.cc is doing? There may be room to put some of that logic in the base Shader class.

Reply:
Sorry for being unclear. I was only noting that the TODO needed an issue filed for it.

Reply:
Perfect, thanks!